tests/unit-tests/test-class-posttypes.php

```diff
@@ -271,4 +271,40 @@ public function testFireNoActionOnExistingCourseUpdate() {
 		Sensei()->post_types->fire_scheduled_initial_publish_actions();
 		$this->assertEquals( 1, did_action( 'sensei_course_initial_publish' ) );
 	}
+
+	/**
+	 * Test if the query vars rewrite appropriately when the page id and the course page id matches
+	 *
+	 * @covers Sensei_PostType::handle_course_archive_page
+	 */
+	public function testHandleCourseArchivePageShouldRewriteTheUrlAppropriately() {
+		Sensei()->settings->set( 'course_page', '1' );
+		$request = [
+			'page_id' => '1',
+			'paged'   => '5',
+		];
+		$request = Sensei()->post_types->handle_course_archive_page( $request );
```

Comment:
One question: Should I call this method directly or should I use `apply_filters('request', $request');` instead?

They have basically the same effect according to my tests (as `handle_course_archive_page` is registered using `add_filter` anyway), but I didn't found what's the best practice here..